[Model Element] Update entity load method call

achan00 opened the pull request:
#### 5a0f853f6f06f0203f51a3d9da2a0d3f6950deb0
<pre>
[Model Element] Update entity load method call
<a href="https://bugs.webkit.org/show_bug.cgi?id=293976">https://bugs.webkit.org/show_bug.cgi?id=293976</a>
<a href="https://rdar.apple.com/150533960">rdar://150533960</a>

Reviewed by Richard Robinson.

Switch to Entity(from:named:). Also, simplify WKRKEntity.applyDefaultIBL()
to not use a Task since EnvironmentResource.defaultObject() should be safe
to execute on main thread.

* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.swift:
(WKRKEntity.load(from:withAttributionTaskID:completionHandler:)):
(WKRKEntity.applyDefaultIBL):

Canonical link: <a href="https://commits.webkit.org/295788@main">https://commits.webkit.org/295788@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/641af5a629c954b0b2f5a1ec2dbaec2a969a4ff3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106097 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/25846 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16240 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111294 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56693 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26502 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34349 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80591 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109101 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/20934 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95741 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60913 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20511 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56131 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/90300 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/13874 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114149 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33235 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/24523 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/89696 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33599 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/91969 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89360 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22791 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34224 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12036 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/28808 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33160 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/38572 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/32906 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36256 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34504 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.swift:
```diff
@@ -60,16 +60,14 @@ extension WKRKEntity {
     @objc(loadFromData:withAttributionTaskID:completionHandler:)
     class func load(from data: Data, withAttributionTaskID attributionTaskId: String?, completionHandler: @MainActor @Sendable @escaping (WKRKEntity?) -> Void) {
-#if canImport(RealityKit, _version: 377)
+#if canImport(RealityKit, _version: "403.0.3")
```
why is this a string, when it wasn't before?

achan00 replied:
I was following other `canImport` checks in WebKit where dotted version strings were quoted. Either one should work.

achan00 committed:
[Model Element] Update entity load method call
https://bugs.webkit.org/show_bug.cgi?id=293976
rdar://150533960

Reviewed by Richard Robinson.

Switch to Entity(from:named:). Also, simplify WKRKEntity.applyDefaultIBL()
to not use a Task since EnvironmentResource.defaultObject() should be safe
to execute on main thread.

* Source/WebKit/WebKitSwift/RealityKit/WKRKEntity.swift:
(WKRKEntity.load(from:withAttributionTaskID:completionHandler:)):
(WKRKEntity.applyDefaultIBL):

Canonical link: https://commits.webkit.org/295788@main
